src/assets/VueOptionVsCompositionAPI/option-vs-composition-api.md

```diff
@@ -0,0 +1,186 @@
+---
+title: Option Vs Composition API
+authorName: Sagar Gurung
+authorAvatar: https://avatars.githubusercontent.com/u/46086950?v=4
+authorLink: https://github.com/SagarGi
+createdAt: Dec 13, 2023
+tags: vue, option API, composition API, framework, frontend
+banner: https://blog.jankaritech.com/src/assets/DockerCompose/images/OptionVsCompositionAPI.png
+---
+
+## Background
+When it comes to defining or creating a component in the `Vue.js`, we have heard about the `Option API`. We can create specific aspects or behaviors of a component and how we want to build through it.
+It provides different properties or options such as `data`, `methods`, `computed`, and `watch` through which we can organize the behavior of a component systematically. But since `Vue 3`, we have
+another API called `Composition API` through which we can achieve the same thing. The `Composition API` is a new highlight for many vue developers but both approaches to define a component have their
+unique pros and cons. So debating which is best among these two would be a difficult one.
+
+In this blog, we will discuss how we can convert a component defined in `Option API` to `Composition API`. Also at the end, we will know how we can define the component with `Composition API`.
+I have already created a simple application [Voting Details](https://github.com/SagarGi/OptionsVsCompositionVue). We will simply be examining it for the blog.
+
+## Play around with `Voting Details`
+I have put this section in this blog since it's better to know some information about what we are building. This demo application simply stores the user's details and checks whether a user is eligible to vote are not.
```

Comment:
```suggestion
I have put this section in this blog since it's better to know some information about what we are building. This demo application simply stores the user's details and checks whether a user is eligible to vote or not.
```